### PR TITLE
[script.service.singealarmclock] 1.01

### PR DESCRIPTION
**Pull request by druscoe**
### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [script.foo.bar] 1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: script.service.singealarmclock/addon.xml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<addon id="script.service.singealarmclock"
+       name="Alarm Clock"
+       version="1.01"
```
Comment (enen92) [inline script.service.singealarmclock/addon.xml]: please use major.minor.revision for the version format

File: script.service.singealarmclock/addon.xml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+       provider-name="Damien Ruscoe">
+  <requires>
+    <import addon="xbmc.python" version="2.1.0"/>
```
Comment (enen92) [inline script.service.singealarmclock/addon.xml]: 2.26.0

File: script.service.singealarmclock/addon.xml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  </requires>
+
+  <extension point="xbmc.service" library="default.py" />
+
+  <extension point="xbmc.python.pluginsource" library="plugin.py">
+    <provides>executable</provides>
+  </extension>
+
+  <extension point="xbmc.addon.metadata">
+    <summary lang="en">Alarm Clock</summary>
```
Comment (enen92) [inline script.service.singealarmclock/addon.xml]: en_GB

File: script.service.singealarmclock/addon.xml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    <description lang="en">Schedule custom alarms and wake up to your favourite media.</description>
```
Comment (enen92) [inline script.service.singealarmclock/addon.xml]: en_GB

File: script.service.singealarmclock/addon.xml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+    <website>http://damienruscoe.co.uk</website>
+    <forum>https://bitbucket.org/Chuddah/script.service.singealarmclock/wiki/Home</forum>
+    <source>https://bitbucket.org/Chuddah/script.service.singealarmclock/src/master</source>
+    <email>chuddah@gmail.com</email>
+    <news>Watch this space!</news>
```
Comment (enen92) [inline script.service.singealarmclock/addon.xml]: please use proper revision information

File: script.service.singealarmclock/addon.xml [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  <extension point="xbmc.python.pluginsource" library="plugin.py">
+    <provides>executable</provides>
+  </extension>
+
+  <extension point="xbmc.addon.metadata">
+    <summary lang="en">Alarm Clock</summary>
+    <description lang="en">Schedule custom alarms and wake up to your favourite media.</description>
+
+    <website>http://damienruscoe.co.uk</website>
+    <forum>https://bitbucket.org/Chuddah/script.service.singealarmclock/wiki/Home</forum>
```
Comment (enen92) [inline script.service.singealarmclock/addon.xml]: this is for kodi.tv forum thread if available. If not please drop.

File: script.service.singealarmclock/addon.xml [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    <source>https://bitbucket.org/Chuddah/script.service.singealarmclock/src/master</source>
+    <email>chuddah@gmail.com</email>
+    <news>Watch this space!</news>
+
+    <platform>all</platform>
+    <license>GNU GENERAL PUBLIC LICENSE Version 2, June 1991</license>
```
Comment (enen92) [inline script.service.singealarmclock/addon.xml]: Please use a spdx identifier for the license: see https://spdx.org/licenses/

File: script.service.singealarmclock/addon.xml [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    <language />
```
Comment (enen92) [inline script.service.singealarmclock/addon.xml]: please drop language tag

File: script.service.singealarmclock/changelog.txt [hunk 9]
```diff
@@ -0,0 +1,2 @@
+v0.0.42
```
Comment (enen92) [inline script.service.singealarmclock/changelog.txt]: please drop, no longer parsed by kodi in favor of the news tag in addon .xml

File: script.service.singealarmclock/default.py [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+
+
+class KodiQuartz(CronTab.Quartz):
+    def __init__(self, monitor):
+        super(KodiQuartz, self).__init__()
+        self.monitor = monitor
+
+    def is_still_alive(self):
+        return super(KodiQuartz, self).is_still_alive() \
+                and not xbmc.abortRequested
```
Comment (enen92) [inline script.service.singealarmclock/default.py]: `xbmc.Monitor().abortRequested()`

File: script.service.singealarmclock/default.py [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+    speakers = Speakers()
+    alarmClock = AlarmClock(crontab, speakers)
+    monitor = AlarmClockMonitor(alarmClock)
+
+    quartz = KodiQuartz(monitor)
+    crontab.quartz = quartz
+
+    alarmClock.crontab.start()
+
+    logger.info('Ended service' + ' '.join(sys.argv))
```
Comment (enen92) [inline script.service.singealarmclock/default.py]: use debug for most of the log calls. Also configure logging to use xbmc.log. See an example here: https://github.com/xbmc/generator-kodi-addon/blob/master/generators/app/templates/resources/lib/kodilogging.py

File: script.service.singealarmclock/plugin.py [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+#    along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+#    This script is based on service.skin.widgets
+#    Thanks to the original authors
+
+import xbmc, xbmcaddon
+import os, sys
+__cwd__      = xbmc.translatePath( xbmcaddon.Addon().getAddonInfo('path') ).decode("utf-8")
+__resource__ = xbmc.translatePath( os.path.join( __cwd__, 'resources', 'lib' ) ).decode("utf-8")
+sys.path.append(__resource__)
```
Comment (enen92) [inline script.service.singealarmclock/plugin.py]: bad practice to use sys.path.append. Why do you need it? Can't you just import your python code as modules?

File: script.service.singealarmclock/default.py [hunk 13]
```diff
@@ -0,0 +1,141 @@
+import xbmc, xbmcaddon
+import os, sys
+__cwd__      = xbmc.translatePath( xbmcaddon.Addon().getAddonInfo('path') ).decode("utf-8")
+__resource__ = xbmc.translatePath( os.path.join( __cwd__, 'resources', 'lib' ) ).decode("utf-8")
+sys.path.append(__resource__)
```
Comment (enen92) [inline script.service.singealarmclock/default.py]: bad practice. Can't you just import the modules you need?

File: script.service.singealarmclock/plugin.py [hunk 14]
```diff
@@ -0,0 +1,131 @@
...
+    pairs = ['%s=%s' % kv for kv in list(query_values.items())]
+    return '?' + '&'.join(pairs)
+
+
+def menu_item(plugin, label, **query_values):
+    list_item = xbmcgui.ListItem(label)
+    list_item.setLabel(label)
+    list_item.setLabel2(label)
+    list_item.setProperty('IsPlayable', 'false')
+    list_item.setThumbnailImage(__cwd__ + '/icon.png')
```
Comment (enen92) [inline script.service.singealarmclock/plugin.py]: drop `setThumbnailImage` and `setIconImage`, they'll result in NOP now in matrix

File: script.service.singealarmclock/plugin.py [hunk 15]
```diff
@@ -0,0 +1,131 @@
...
+    list_item.setIconImage(__cwd__ + '/icon.png')
+    list_item.setArt(__cwd__ + '/fanart.jpg')
```
Comment (enen92) [inline script.service.singealarmclock/plugin.py]: `ListItem.setArt` expects a dict containing all the art...
`list_item.setArt({'thumb':'xxx', 'fanart': xxxx, 'icon: 'xxx' ... })`

File: script.service.singealarmclock/plugin.py [hunk 16]
```diff
@@ -0,0 +1,131 @@
...
+
+    params = dict(urlparse.parse_qsl(sys.argv[2].lstrip('?')))
+    plugin = sys.argv[0]
+    items = []
+
+    entering_subfolder = True
+    if params.get('type') is None:
+        if not settings.getAlarms():
+            settings.edit_new_alarm()
+        items.append(menu_item(plugin, 'Add Alarm',         type='add_alarm'))
```
Comment (enen92) [inline script.service.singealarmclock/plugin.py]: all these strings must be localized as they are presented to the user in the UI

File: script.service.singealarmclock/resources/lib/tests/test_alarmclock.py [hunk 17]
```diff
@@ -0,0 +1,141 @@
+# pylint: disable=missing-docstring,invalid-name,import-error
```
Comment (enen92) [inline script.service.singealarmclock/resources/lib/tests/test_alarmclock.py]: please remove `/tests` from the submission. They are useful for development but not for distribution.

File: script.service.singealarmclock/settings.py [hunk 18]
```diff
@@ -0,0 +1,177 @@
...
+
+
+def edit_alarm(alarm_id):
+    addon = __create_addon()
+    __edit_alarm(addon, alarm_id)
+
+
+def edit_new_alarm():
+    addon = __create_addon()
+    default_stream = "http://media-ice.musicradio.com:80/HeartBristolMP3.m3u"
```
Comment (enen92) [inline script.service.singealarmclock/settings.py]: intentional?

**Comment by enen92** [comment]
ping @druscoe

**Comment by basrieter** [comment]
last ping @druscoe

**Comment by basrieter** [comment]
Closing this PR due to inactivity. Please re-open it whenever you have time to look at our suggestions.

**Pull request closed without merging**